Added Purpose of chunks (#13)
mqtt-varroa/varroa-documentation: LaTeX/Parts/Chapter-Chunk-Concept.tex

\chapter{Chunk Concept}
\section{Definition}

> A chunk is a type of container designed to distribute data from the Commander to its Agents.

Reviewer: ```suggestion
A chunk is a work package of the Scenario execution designed to be distributed from the Commander to its Agents.
```

Reviewer: It is then handled by the Agent it is distributed to.
The Agent now creates or uses existing MQTT Agents to execute the Chunk.

> It contains:

Reviewer: maybe specify the different forms in which chunks can exist

Author: ?

\begin{itemize}
	\item the amount of clients a chunk represents.

> 	\item the commands that are applied to the clients in the chunk.

Reviewer: ```suggestion
	\item the commands that are executed by the clients
```

	\item information concerning the MQTT-specific properties.

> 	\item variables defining lifecycle properties.

Reviewer: too generic, specify the variables and their values

\end{itemize}

> \section{Purpose of Chunks}

Reviewer: We do not need to describbe why we use chunks, we need to explain the concept itself.

Due to the distributed nature of Varroa it was necessary to be able to spread the information contained in the scenario.

> To secure a deterministic distribution of a scenario's data the chunks had to be evenly spread across the initialized Agents.

Reviewer: factually incorrect. Even spreading of the chunks does not induce determinism. Determinism means that every chunk is distributed to the exact same Agent in every Varroa Execution.

> To ensure this determinism a mechanism was introduced combining a round-robin selection of chunks with a handshake mechanism of the agents.

Reviewer: 👍
